addons/web/controllers/main.py

```diff
@@ -924,6 +924,21 @@ def web_login(self, redirect=None, **kw):
         response.headers['X-Frame-Options'] = 'DENY'
         return response
 
+    # sid changes once connected, therefore csrf changes too.
+    @http.route('/web/auth/api/request_key', type='http', auth="user", methods=['GET'], website=True)
```

Comment:
It woudn't be the first time we put ad-hoc stuff in the core out of convenience, but at the same time there *is* truth to al's objection: we don't really know that this will work for anything other than outlook right now, so maybe this should be moved (back) to the relevant module, to be used (updated?) if suitable when gmail or whatever gets integrated, and at that point deprecated (or or removed, I don't know how the add-ins are synchronised) over there and integrated into the web core?